lib/handlers/revoke-handler.ts

```diff
@@ -0,0 +1,338 @@
+import * as auth from 'basic-auth';
+import {
+  InvalidArgumentError,
+  InvalidClientError,
+  InvalidRequestError,
+  InvalidTokenError,
+  OAuthError,
+  ServerError,
+} from '../errors';
+import { Client, Model } from '../interfaces';
+import { Request } from '../request';
+import { Response } from '../response';
+import { oneSuccess } from '../utils/fn';
+import * as is from '../validator/is';
+
+export class RevokeHandler {
+  model: Model;
+  constructor(options: any = {}) {
+    if (!options.model) {
+      throw new InvalidArgumentError('Missing parameter: `model`');
+    }
+
+    if (!options.model.getClient) {
+      throw new InvalidArgumentError(
+        'Invalid argument: model does not implement `getClient()`',
+      );
+    }
+
+    if (!options.model.getRefreshToken) {
+      throw new InvalidArgumentError(
+        'Invalid argument: model does not implement `getRefreshToken()`',
+      );
+    }
+
+    if (!options.model.getAccessToken) {
+      throw new InvalidArgumentError(
+        'Invalid argument: model does not implement `getAccessToken()`',
+      );
+    }
+
+    if (!options.model.revokeToken) {
+      throw new InvalidArgumentError(
+        'Invalid argument: model does not implement `revokeToken()`',
+      );
+    }
+
+    this.model = options.model;
+  }
+
+  /**
+   * Revoke Handler.
+   */
+
+  async handle(request: Request, response: Response) {
+    if (!(request instanceof Request)) {
+      throw new InvalidArgumentError(
+        'Invalid argument: `request` must be an instance of Request',
+      );
+    }
+
+    if (!(response instanceof Response)) {
+      throw new InvalidArgumentError(
+        'Invalid argument: `response` must be an instance of Response',
+      );
+    }
+
+    if (request.method !== 'POST') {
+      throw new InvalidRequestError('Invalid request: method must be POST');
+    }
+
+    if (!request.is('application/x-www-form-urlencoded')) {
+      throw new InvalidRequestError(
+        'Invalid request: content must be application/x-www-form-urlencoded',
+      );
+    }
+    try {
```

Comment:
`this.model.request = request;`
Should be added too like other handlers.

Reply:
yes, i also think the same, but it's not in the dev branch

Reply:
It has been merged in #462 when revoke-handler was not a part of the code. I think who merged revoke-handler forgot to check for this line.

Reply:
yes, i think you are right, i should must be added